core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -80,7 +88,8 @@ public static boolean isIngestNode(Settings settings) {
     }
 
     private final String nodeName;
-    private final String nodeId;
+    private final String processId;
+    private final String persistentNodeId;
```

Comment:
Can we just get away with calling it `nodeId`? We don't have non-persistent node IDs. Yes it's slightly confusing in the transition from what use to be `nodeId` and is now `processId`, but I think in the long-run it would be better if we just called it `nodeId`?